assets/src/hooks/usePersistedStateReducer.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useReducer } from "react"
+import { loadState, saveState } from "../localStorage"
+import { Dispatch, Reducer, State } from "../state"
+
+const usePersistedStateReducer = (
+  reducer: Reducer,
```

Comment:
This hook and `localStorage.ts` seem to be halfway between trying to be generic/reusable and being specific to the app state. E.g. the hook takes a `Reducer` as an argument, but there's only one function that could possibly fit that spec, `state.reducer`.

I'd go all or nothing. Either pass everything in (including `saved_keys` and `app_state_key`), and make the types generic, or make everything specific, where the hook has no arguments and is only usable by the top level `<App>`

Making everything specific would probably be easier. But if this hook becomes truly reusable, perhaps we could store the reversal states by giving each route its own key instead of by moving the reversal states to the top level (this may not be valuable, since we want to move the reversal state up anyway for other reasons).

Reply:
Well, one big goal is to have the same format as `useReducer`. But this comment and the one below are making me think that `APP_STATE_KEY` and `SAVED_KEYS` should be owned by `usePersistedStateReducer` and `localStorage` should be a naive repository.

Reply:
Ah, yeah, `localStorage.ts` is looking pretty nice now. 👍 

I don't think having the exact same interface as `useReducer` is necessary (despite complimenting it earlier) if we're not planning on having it be a drop in replacement (which it can't be when this file has hardcoded references to `APP_STATE_KEY`, `PERSISTED_KEYS`, and the `State` and `Reducer` types).